### PR TITLE
Simplify sql normalization

### PR DESCRIPTION
**Pull request by richardstartin**
Simplifies the normalisation code because it is too complicated, also fixes an edge case with literals like `','`

### CODE REVIEW

File: internal-api/src/test/groovy/datadog/trace/api/normalize/SQLNormalizerTest.groovy [hunk 1]
```diff
@@ -11,69 +11,74 @@ class SQLNormalizerTest extends DDSpecification {
     then:
     normalized as String == expected
     where:
-    sql                                                                                                                              | expected
```
Comment (richardstartin) [inline internal-api/src/test/groovy/datadog/trace/api/normalize/SQLNormalizerTest.groovy:14]: I don't envy anyone reading this diff, but no query has been edited, a few have been added at the start. Spock is hostile to maintaining git blame.

Comment (bantonsson) [reply internal-api/src/test/groovy/datadog/trace/api/normalize/SQLNormalizerTest.groovy:14]: This looks a lot like `spotless` going crazy, and I would have added `// spotless:off` `// spotless:on` around the table.

**Review by tylerbenson — APPROVED** [review]
Just going to trust you on this one.

File: internal-api/src/main/java/datadog/trace/api/normalize/SQLNormalizer.java [hunk 2]
```diff
@@ -21,150 +20,97 @@
...
   }
 
-  private static long findSplitters(long word) {
-    return tag(word, SPACES)
-        | tag(word, TABS)
-        | tag(word, NEW_LINES)
-        | tag(word, L_PAREN)
-        | tag(word, R_PAREN)
-        | tag(word, COMMAS);
+  private static BitSet findSplitterPositions(byte[] utf8) {
```
Comment (bantonsson) [inline internal-api/src/main/java/datadog/trace/api/normalize/SQLNormalizer.java:93]: This logic is soo much easier to follow 👍

**Review by bantonsson — APPROVED** [review]

File: internal-api/src/test/groovy/datadog/trace/api/normalize/SQLNormalizerTest.groovy (hunk 1 above)
Comment (richardstartin) [reply internal-api/src/test/groovy/datadog/trace/api/normalize/SQLNormalizerTest.groovy:14]: I've added that but I think this table is scorched earth now.

File: internal-api/src/main/java/datadog/trace/api/normalize/SQLNormalizer.java (hunk 2 above)
Comment (richardstartin) [reply internal-api/src/main/java/datadog/trace/api/normalize/SQLNormalizer.java:101]: sorry it was so messy before

File: internal-api/src/test/groovy/datadog/trace/api/normalize/SQLNormalizerTest.groovy (hunk 1 above)
Comment (bantonsson) [reply internal-api/src/test/groovy/datadog/trace/api/normalize/SQLNormalizerTest.groovy:14]: Not important, but you could salvage it if you want:
```
# copy your new test cases
> git reset --soft HEAD~1
> git reset internal-api/src/test/groovy/datadog/trace/api/normalize/SQLNormalizerTest.groovy
> git checkout -- internal-api/src/test/groovy/datadog/trace/api/normalize/SQLNormalizerTest.groovy
# paste in the off/on and your new test cases
```

**Pull request merged**